src/test/archive/patch_archive_test/patch_archive_test.ml

```diff
@@ -80,33 +62,21 @@ let main ~db_uri ~network_data_folder () =
         (* never remove last block as missing-block-guardian can have issues when patching it
            as it patching only gaps
         *)
-        Random.int (List.length extensional_files - 1) )
+        Random.int (List.length extensional_files - 2) + 1 )
   in
 
   let unpatched_extensional_files =
     List.filteri extensional_files ~f:(fun i _ ->
         not (List.mem n i ~equal:Int.equal) )
-    |> List.dedup_and_sort ~compare:(fun left right ->
-           let scan_height item =
-             let item =
-               Filename.basename item |> Str.global_replace (Str.regexp "-") " "
-             in
-             Scanf.sscanf item "%s %d %s" (fun _ height _ -> height)
-           in
-
-           let left_height = scan_height left in
-           let right_height = scan_height right in
-
-           Int.compare left_height right_height )
+    |> Utils.sort_archive_files
```

Comment:
Am I missing something?

![image](https://github.com/user-attachments/assets/ded456f5-a8f0-4bfa-9fd7-5bcdc7c56b64)

Reply:
Where did deduplication gone? It's not in `Utils.sort_archive_files`

Reply:
Another good catch. Adding dedup to sort_archive_files function